cc3d/player5/Plugins/ViewManagerPlugins/DemoBrowser.py

```diff
@@ -12,9 +12,228 @@
+
+            self.selectedPath = absPath
+
+            parentDir = absPath.parent
+
+            description = self.getFilePreview(parentDir, DESCRIPTION_FNAME)
+            if description == MISSING_FILE_MESSAGE:
+                self.descriptionText.setPlainText(str(absPath))
+            else:
+                self.descriptionText.setPlainText(str(absPath) + "\n\n" + description)
+
+            self.pythonPreviewText.setPlainText(self.getFilePreview(parentDir, "*Steppables.py"))
+            self.pythonHighlighter.highlightBlock(self.pythonPreviewText.toPlainText())
+
+            self.xmlPreviewText.setPlainText(self.getFilePreview(parentDir, "*.xml"))
+
+            self.openDemoButton.show()
+            self.demoTabView.show()
+
+            # itemData is a dictionary (for some reason), and the text is at key 0
+            demoName = self.demoListView.model().itemData(selectedIndex)[0]
+            self.selectedDemoNameLabel.setText(demoName)
+            self.selectedDemoNameLabel.show()
+
+            self.noSelectionPlaceholder.hide()
+        except KeyError:
+            self.showDemoOpenError()
+            print("KeyError occurred while choosing a demo.")
+        except Exception as ex:
+            self.showDemoOpenError()
+            print("Error: Could not select that demo. Exception:\n", ex)
+
+    def showDemoOpenError(self):
+        QMessageBox.warning(
+            None,
+            "Warning",
+            "We could not select that demo because of an error. "
+            + "Please close the Demo Browser, then try again. "
+            + "If the issue persists, please delete the .demoindex "
+            + "file inside of your demos folder. ",
+            QMessageBox.Ok,
+        )
+
+    def openDemo(self):
+        # Sanity check.
+        # The button to trigger this should be invisible anyway.
+        if not self.selectedPath or not self.selectedPath.exists():
+            self.showDemoOpenError()
+            print("Error: Could not open the demo with path", self.selectedPath)
+            return
+
+        if self.simpleTabView():
+            self.simpleTabView().openSim(str(self.selectedPath))
+            self.close()
+
+
+class CreateIndexWorker(QThread):
+    finished = pyqtSignal()
+
+    def __init__(self, parent=None, *args, **kwargs):
+        super().__init__(parent=parent, *args, **kwargs)
+        self.searchIndex = None
+
+    def run(self):
+        # Simulate a long-running task
+        # self.searchIndex = createOrLoadSearchIndex()
+        import time
+        time.sleep(5)
+        self.finished.emit()  # Emit finished signal when task is done
+
+
+class CreateIndexWaitPopup(QMessageBox):
+    def __init__(self, parent=None):
+        super(CreateIndexWaitPopup, self).__init__(parent)
+        self.setWindowModality(Qt.ApplicationModal)
+        self.setWindowTitle('Generating Demos Index - Please Wait')
+        self.setGeometry(100, 100, 500, 100)
```

Comment:
Can you make it a little wider? It comes up as "Generating Demos Ind..." for me.